iam-login-service/src/main/java/it/infn/mw/iam/core/oauth/profile/wlcg/WLCGProfileAccessTokenBuilder.java

```diff
@@ -119,9 +121,29 @@ private void addAuthnInfoClaims(Builder builder, Set<String> scopes, UserInfo us
   }
 
   private void addAudience(Builder builder, OAuth2Authentication authentication) {
-    if (!hasAudienceRequest(authentication) && !hasRefreshTokenAudienceRequest(authentication)) {
+    if (!hasAnyAudience(authentication)) {
       builder.audience(ALL_AUDIENCES_VALUE);
```

Comment:
we could check if builder.getClaims("aud") has been already defined. In case not, we must set it as the "any" value